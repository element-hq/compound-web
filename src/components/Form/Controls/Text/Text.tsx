/*
Copyright 2023 The Matrix.org Foundation C.I.C.
Copyright 2023 New Vector Ltd

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { forwardRef, ComponentProps, ComponentRef } from "react";
import { Control } from "@radix-ui/react-form";

import styles from "./Text.module.css";
import classNames from "classnames";

type TextProps = {
  /**
   * The CSS class name.
   */
  className?: string;

  /**
   * Enable contextual alternate ligatures on input text
   * For example on an in-place editing field
   * https://github.com/rsms/inter/issues/222
   * https://github.com/rsms/inter/blob/master/src/features/calt.fea
   */
  enableLigatures?: boolean;
} & ComponentProps<"input">;

/**
 * A styled text input, for standalone use.
 */
export const TextInput = forwardRef<HTMLInputElement, TextProps>(
  function TextInput(props, ref) {
    const classes = classNames(
      styles.control,
      props.className,
      props.enableLigatures && styles["enable-ligatures"],
    );
    return <input ref={ref} {...props} className={classes} />;
  },
);

/**
 * A styled text input wrapped in a `Control` component, for use in Radix forms.
 */
export const TextControl = forwardRef<
  ComponentRef<typeof TextInput>,
  ComponentProps<typeof TextInput>
>(function TextControl(props, ref) {
  return (
    <Control asChild>
      <TextInput ref={ref} {...props} />
    </Control>
  );
});
