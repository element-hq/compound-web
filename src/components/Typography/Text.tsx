/*
Copyright 2025 New Vector Ltd.
Copyright 2023 The Matrix.org Foundation C.I.C.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import React from "react";
import { Typography } from "./Typography";

type TypographyProps = React.ComponentProps<typeof Typography>;

/**
 * A text component. Underlying HTML element can be changed using the `as`
 * property. Will default to be a paragraph.
 */
export const Text: React.FC<
  Omit<TypographyProps, "type"> & {
    // xl is not a valid text size
    size?: Exclude<TypographyProps["size"], "xl">;
  }
> = ({ as = "p", children, ...props }) => {
  return (
    <Typography as={as} type="body" {...props}>
      {children}
    </Typography>
  );
};
