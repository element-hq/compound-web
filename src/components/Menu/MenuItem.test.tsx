/*
Copyright 2023 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
Please see LICENSE files in the repository root for full details.
*/

import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";
import LeaveIcon from "@vector-im/compound-design-tokens/assets/web/icons/leave";
import UserProfileIcon from "@vector-im/compound-design-tokens/assets/web/icons/user-profile";
import MicOnIcon from "@vector-im/compound-design-tokens/assets/web/icons/mic-on";

import { MenuItem } from "./MenuItem";
import { Text } from "../Typography/Text";

import * as stories from "./MenuItem.stories";
import { composeStories } from "@storybook/react";

const {
  Primary,
  Critical,
  PrimaryDisabled,
  CriticalDisabled,
  IconIsComponent,
  WithALongLabel,
  WithALongLabelAndChildren,
  WithoutChevron,
  WithoutIcon,
} = composeStories(stories);

describe("MenuItem", () => {
  it("renders", () => {
    const { asFragment } = render(
      <MenuItem
        kind="critical"
        Icon={LeaveIcon}
        label="Leave room"
        onSelect={() => {}}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with a child", () => {
    const { asFragment } = render(
      <MenuItem Icon={UserProfileIcon} label="People" onSelect={() => {}}>
        <Text as="span" size="sm">
          10
        </Text>
      </MenuItem>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without a label", () => {
    const { asFragment } = render(
      <MenuItem Icon={MicOnIcon} label={null} onSelect={() => {}}>
        Imagine that there might be a volume slider here in place of the label
      </MenuItem>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with an component as an Icon", () => {
    const { asFragment } = render(<IconIsComponent />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a Primary menu item", () => {
    const { container } = render(<Primary />);
    expect(container).toMatchSnapshot();
  });
  it("renders a Critical menu item", () => {
    const { container } = render(<Critical />);
    expect(container).toMatchSnapshot();
  });
  it("renders a Primary Disabled menu item", () => {
    const { container } = render(<PrimaryDisabled />);
    expect(container).toMatchSnapshot();
  });
  it("renders a Critical Disabled menu item", () => {
    const { container } = render(<CriticalDisabled />);
    expect(container).toMatchSnapshot();
  });
  it("renders a menu item with a long label", () => {
    const { container } = render(<WithALongLabel />);
    expect(container).toMatchSnapshot();
  });
  it("renders a menu item with a long label and children", () => {
    const { container } = render(<WithALongLabelAndChildren />);
    expect(container).toMatchSnapshot();
  });
  it("renders a menu item without a chevron", () => {
    const { container } = render(<WithoutChevron />);
    expect(container).toMatchSnapshot();
  });
  it("renders a menu item without an icon", () => {
    const { container } = render(<WithoutIcon />);
    expect(container).toMatchSnapshot();
  });
});
