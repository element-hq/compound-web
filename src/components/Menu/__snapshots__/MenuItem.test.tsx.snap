// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuItem > renders 1`] = `
<DocumentFragment>
  <button
    class="_item_831119 _interactive_831119"
    data-kind="critical"
    role="menuitem"
  >
    <svg
      aria-hidden="true"
      class="_icon_831119"
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 13q.424 0 .713-.287A.97.97 0 0 0 15 12a.97.97 0 0 0-.287-.713A.97.97 0 0 0 14 11a.97.97 0 0 0-.713.287A.97.97 0 0 0 13 12q0 .424.287.713.288.287.713.287"
      />
      <path
        d="M10.385 21.788A1 1 0 0 1 10 21V3a1.003 1.003 0 0 1 1.242-.97l8 2A1 1 0 0 1 20 5v14a1 1 0 0 1-.758.97l-8 2a1 1 0 0 1-.857-.182M18 5.781l-6-1.5v15.438l6-1.5zM9 6H7v12h2v2H7a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2z"
      />
    </svg>
    <span
      class="_typography_489030 _font-body-md-medium_489030 _label_831119"
    >
      Leave room
    </span>
    <svg
      aria-hidden="true"
      class="_nav-hint_831119"
      fill="currentColor"
      height="24"
      viewBox="8 0 8 24"
      width="8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`MenuItem > renders a Critical Disabled menu item 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119 _disabled_831119"
      data-kind="critical"
      disabled=""
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_831119"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1.5 21.25 1.45-4.95a10.2 10.2 0 0 1-.712-2.1A10.2 10.2 0 0 1 2 12q0-2.075.788-3.9a10.1 10.1 0 0 1 2.137-3.175q1.35-1.35 3.175-2.137A9.7 9.7 0 0 1 12 2q2.075 0 3.9.788a10.1 10.1 0 0 1 3.175 2.137q1.35 1.35 2.137 3.175A9.7 9.7 0 0 1 22 12a9.7 9.7 0 0 1-.788 3.9 10.1 10.1 0 0 1-2.137 3.175q-1.35 1.35-3.175 2.137A9.7 9.7 0 0 1 12 22q-1.125 0-2.2-.238a10.2 10.2 0 0 1-2.1-.712L2.75 22.5a.94.94 0 0 1-1-.25.94.94 0 0 1-.25-1m2.45-1.2 3.2-.95a1 1 0 0 1 .275-.062q.15-.013.275-.013.225 0 .438.038.212.036.412.137a7.4 7.4 0 0 0 1.675.6Q11.1 20 12 20q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4 6.325 6.325 4 12q0 .9.2 1.775t.6 1.675q.176.325.188.688t-.088.712z"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Menu item
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders a Critical menu item 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119"
      data-kind="critical"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_831119"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1.5 21.25 1.45-4.95a10.2 10.2 0 0 1-.712-2.1A10.2 10.2 0 0 1 2 12q0-2.075.788-3.9a10.1 10.1 0 0 1 2.137-3.175q1.35-1.35 3.175-2.137A9.7 9.7 0 0 1 12 2q2.075 0 3.9.788a10.1 10.1 0 0 1 3.175 2.137q1.35 1.35 2.137 3.175A9.7 9.7 0 0 1 22 12a9.7 9.7 0 0 1-.788 3.9 10.1 10.1 0 0 1-2.137 3.175q-1.35 1.35-3.175 2.137A9.7 9.7 0 0 1 12 22q-1.125 0-2.2-.238a10.2 10.2 0 0 1-2.1-.712L2.75 22.5a.94.94 0 0 1-1-.25.94.94 0 0 1-.25-1m2.45-1.2 3.2-.95a1 1 0 0 1 .275-.062q.15-.013.275-.013.225 0 .438.038.212.036.412.137a7.4 7.4 0 0 0 1.675.6Q11.1 20 12 20q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4 6.325 6.325 4 12q0 .9.2 1.775t.6 1.675q.176.325.188.688t-.088.712z"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Menu item
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders a Primary Disabled menu item 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119 _disabled_831119"
      data-kind="primary"
      disabled=""
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_831119"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1.5 21.25 1.45-4.95a10.2 10.2 0 0 1-.712-2.1A10.2 10.2 0 0 1 2 12q0-2.075.788-3.9a10.1 10.1 0 0 1 2.137-3.175q1.35-1.35 3.175-2.137A9.7 9.7 0 0 1 12 2q2.075 0 3.9.788a10.1 10.1 0 0 1 3.175 2.137q1.35 1.35 2.137 3.175A9.7 9.7 0 0 1 22 12a9.7 9.7 0 0 1-.788 3.9 10.1 10.1 0 0 1-2.137 3.175q-1.35 1.35-3.175 2.137A9.7 9.7 0 0 1 12 22q-1.125 0-2.2-.238a10.2 10.2 0 0 1-2.1-.712L2.75 22.5a.94.94 0 0 1-1-.25.94.94 0 0 1-.25-1m2.45-1.2 3.2-.95a1 1 0 0 1 .275-.062q.15-.013.275-.013.225 0 .438.038.212.036.412.137a7.4 7.4 0 0 0 1.675.6Q11.1 20 12 20q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4 6.325 6.325 4 12q0 .9.2 1.775t.6 1.675q.176.325.188.688t-.088.712z"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Menu item
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders a Primary menu item 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119"
      data-kind="primary"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_831119"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1.5 21.25 1.45-4.95a10.2 10.2 0 0 1-.712-2.1A10.2 10.2 0 0 1 2 12q0-2.075.788-3.9a10.1 10.1 0 0 1 2.137-3.175q1.35-1.35 3.175-2.137A9.7 9.7 0 0 1 12 2q2.075 0 3.9.788a10.1 10.1 0 0 1 3.175 2.137q1.35 1.35 2.137 3.175A9.7 9.7 0 0 1 22 12a9.7 9.7 0 0 1-.788 3.9 10.1 10.1 0 0 1-2.137 3.175q-1.35 1.35-3.175 2.137A9.7 9.7 0 0 1 12 22q-1.125 0-2.2-.238a10.2 10.2 0 0 1-2.1-.712L2.75 22.5a.94.94 0 0 1-1-.25.94.94 0 0 1-.25-1m2.45-1.2 3.2-.95a1 1 0 0 1 .275-.062q.15-.013.275-.013.225 0 .438.038.212.036.412.137a7.4 7.4 0 0 0 1.675.6Q11.1 20 12 20q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4 6.325 6.325 4 12q0 .9.2 1.775t.6 1.675q.176.325.188.688t-.088.712z"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Menu item
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders a menu item with a long label 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119"
      data-kind="primary"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_831119"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1.5 21.25 1.45-4.95a10.2 10.2 0 0 1-.712-2.1A10.2 10.2 0 0 1 2 12q0-2.075.788-3.9a10.1 10.1 0 0 1 2.137-3.175q1.35-1.35 3.175-2.137A9.7 9.7 0 0 1 12 2q2.075 0 3.9.788a10.1 10.1 0 0 1 3.175 2.137q1.35 1.35 2.137 3.175A9.7 9.7 0 0 1 22 12a9.7 9.7 0 0 1-.788 3.9 10.1 10.1 0 0 1-2.137 3.175q-1.35 1.35-3.175 2.137A9.7 9.7 0 0 1 12 22q-1.125 0-2.2-.238a10.2 10.2 0 0 1-2.1-.712L2.75 22.5a.94.94 0 0 1-1-.25.94.94 0 0 1-.25-1m2.45-1.2 3.2-.95a1 1 0 0 1 .275-.062q.15-.013.275-.013.225 0 .438.038.212.036.412.137a7.4 7.4 0 0 0 1.675.6Q11.1 20 12 20q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4 6.325 6.325 4 12q0 .9.2 1.775t.6 1.675q.176.325.188.688t-.088.712z"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders a menu item with a long label and children 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119"
      data-kind="primary"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_831119"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1.5 21.25 1.45-4.95a10.2 10.2 0 0 1-.712-2.1A10.2 10.2 0 0 1 2 12q0-2.075.788-3.9a10.1 10.1 0 0 1 2.137-3.175q1.35-1.35 3.175-2.137A9.7 9.7 0 0 1 12 2q2.075 0 3.9.788a10.1 10.1 0 0 1 3.175 2.137q1.35 1.35 2.137 3.175A9.7 9.7 0 0 1 22 12a9.7 9.7 0 0 1-.788 3.9 10.1 10.1 0 0 1-2.137 3.175q-1.35 1.35-3.175 2.137A9.7 9.7 0 0 1 12 22q-1.125 0-2.2-.238a10.2 10.2 0 0 1-2.1-.712L2.75 22.5a.94.94 0 0 1-1-.25.94.94 0 0 1-.25-1m2.45-1.2 3.2-.95a1 1 0 0 1 .275-.062q.15-.013.275-.013.225 0 .438.038.212.036.412.137a7.4 7.4 0 0 0 1.675.6Q11.1 20 12 20q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4 6.325 6.325 4 12q0 .9.2 1.775t.6 1.675q.176.325.188.688t-.088.712z"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, empor incididunt ut labore et dolore magna aliqua.
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        Longer children too
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders a menu item without a chevron 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119"
      data-kind="primary"
      role="menuitem"
    >
      <svg
        aria-hidden="true"
        class="_icon_831119"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m1.5 21.25 1.45-4.95a10.2 10.2 0 0 1-.712-2.1A10.2 10.2 0 0 1 2 12q0-2.075.788-3.9a10.1 10.1 0 0 1 2.137-3.175q1.35-1.35 3.175-2.137A9.7 9.7 0 0 1 12 2q2.075 0 3.9.788a10.1 10.1 0 0 1 3.175 2.137q1.35 1.35 2.137 3.175A9.7 9.7 0 0 1 22 12a9.7 9.7 0 0 1-.788 3.9 10.1 10.1 0 0 1-2.137 3.175q-1.35 1.35-3.175 2.137A9.7 9.7 0 0 1 12 22q-1.125 0-2.2-.238a10.2 10.2 0 0 1-2.1-.712L2.75 22.5a.94.94 0 0 1-1-.25.94.94 0 0 1-.25-1m2.45-1.2 3.2-.95a1 1 0 0 1 .275-.062q.15-.013.275-.013.225 0 .438.038.212.036.412.137a7.4 7.4 0 0 0 1.675.6Q11.1 20 12 20q3.35 0 5.675-2.325T20 12t-2.325-5.675T12 4 6.325 6.325 4 12q0 .9.2 1.775t.6 1.675q.176.325.188.688t-.088.712z"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Menu item
      </span>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders a menu item without an icon 1`] = `
<div>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119 _no-icon_831119"
      data-kind="primary"
      role="menuitem"
    >
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Menu item
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</div>
`;

exports[`MenuItem > renders with a child 1`] = `
<DocumentFragment>
  <button
    class="_item_831119 _interactive_831119"
    data-kind="primary"
    role="menuitem"
  >
    <svg
      aria-hidden="true"
      class="_icon_831119"
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.175 13.825Q10.35 15 12 15t2.825-1.175T16 11t-1.175-2.825T12 7 9.175 8.175 8 11t1.175 2.825m4.237-1.412A1.93 1.93 0 0 1 12 13q-.825 0-1.412-.588A1.93 1.93 0 0 1 10 11q0-.825.588-1.412A1.93 1.93 0 0 1 12 9q.825 0 1.412.588Q14 10.175 14 11t-.588 1.412"
      />
      <path
        d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2s10 4.477 10 10m-2 0a8 8 0 1 0-16 0 8 8 0 0 0 16 0"
      />
      <path
        d="M16.23 18.792a13 13 0 0 0-1.455-.455 11.6 11.6 0 0 0-5.55 0q-.73.18-1.455.455a8 8 0 0 1-1.729-1.454q1.336-.618 2.709-.95A13.8 13.8 0 0 1 12 16q1.65 0 3.25.387 1.373.333 2.709.95a8 8 0 0 1-1.73 1.455"
      />
    </svg>
    <span
      class="_typography_489030 _font-body-md-medium_489030 _label_831119"
    >
      People
    </span>
    <svg
      aria-hidden="true"
      class="_nav-hint_831119"
      fill="currentColor"
      height="24"
      viewBox="8 0 8 24"
      width="8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
      />
    </svg>
    <span
      class="_typography_489030 _font-body-sm-regular_489030"
    >
      10
    </span>
  </button>
</DocumentFragment>
`;

exports[`MenuItem > renders with an component as an Icon 1`] = `
<DocumentFragment>
  <div
    style="width: 300px;"
  >
    <button
      class="_item_831119 _interactive_831119"
      data-kind="primary"
      role="menuitem"
    >
      <div
        class="_icon_831119"
        style="height: 24px; width: 48px; background-color: teal;"
      />
      <span
        class="_typography_489030 _font-body-md-medium_489030 _label_831119"
      >
        Menu item
      </span>
      <svg
        aria-hidden="true"
        class="_nav-hint_831119"
        fill="currentColor"
        height="24"
        viewBox="8 0 8 24"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
        />
      </svg>
      <span
        class="_typography_489030 _font-body-sm-regular_489030"
      >
        99
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`MenuItem > renders without a label 1`] = `
<DocumentFragment>
  <button
    class="_item_831119 _interactive_831119 _no-label_831119"
    data-kind="primary"
    role="menuitem"
  >
    <svg
      aria-hidden="true"
      class="_icon_831119"
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 12a1 1 0 1 0-2 0 8 8 0 0 0 7 7.938V21a1 1 0 1 0 2 0v-1.062A8 8 0 0 0 20 12a1 1 0 1 0-2 0 6 6 0 0 1-12 0"
      />
      <path
        clip-rule="evenodd"
        d="M14 12V6a2 2 0 1 0-4 0v6a2 2 0 1 0 4 0M12 2a4 4 0 0 0-4 4v6a4 4 0 0 0 8 0V6a4 4 0 0 0-4-4"
        fill-rule="evenodd"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="_nav-hint_831119"
      fill="currentColor"
      height="24"
      viewBox="8 0 8 24"
      width="8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8.7 17.3a.95.95 0 0 1-.275-.7q0-.425.275-.7l3.9-3.9-3.9-3.9a.95.95 0 0 1-.275-.7q0-.425.275-.7a.95.95 0 0 1 .7-.275q.425 0 .7.275l4.6 4.6q.15.15.213.325.062.175.062.375t-.062.375a.9.9 0 0 1-.213.325l-4.6 4.6a.95.95 0 0 1-.7.275.95.95 0 0 1-.7-.275"
      />
    </svg>
    Imagine that there might be a volume slider here in place of the label
  </button>
</DocumentFragment>
`;
